In [100]:
# This Problem relates to calculating probabilities

In [ ]:
import pandas as pd

In [37]:
filename1 = 'Data/311_Service_Requests_-_Vacant_and_Abandoned_Buildings_Reported'
filename2 = 'Data/311_Service_Requests_-_Graffiti_Removal'
filename3 = 'Data/311_Service_Requests_-_Pot_Holes_Reported'
filename4 = 'Data/311_Service_Requests_-_Sanitation_Code_Complaints'

df1 = pd.read_csv(filename1 + '.csv', skiprows = [1], encoding = "ISO-8859-1")
df2 = pd.read_csv(filename2 + '.csv', encoding = "ISO-8859-1")
df3 = pd.read_csv(filename3 + '.csv', encoding = "ISO-8859-1")
df4 = pd.read_csv(filename4 + '.csv', skiprows = [1], encoding = "ISO-8859-1")

Question 3(a)

In [36]:
df1['STREET ADDRESS'] = df1['ADDRESS STREET NUMBER'].map(str) + ' ' + df1['ADDRESS STREET DIRECTION'].map(str) + ' ' + df1['ADDRESS STREET NAME'].map(str) + ' ' + df1['ADDRESS STREET SUFFIX'].map(str)
df1['STREET ADDRESS'][df1['STREET ADDRESS'] == '7500 S WOLCOTT AVE'] # no complaint

Series([], Name: STREET ADDRESS, dtype: object)

In [35]:
df2['Street Address'] = df2['Street Address'].astype('str') 
df2['Street Address'][df2['Street Address'] == '7500 S WOLCOTT AVE'] # no complaint

Series([], Name: Street Address, dtype: object)

In [25]:
df3['STREET ADDRESS'][df3['STREET ADDRESS'] == '7500 S WOLCOTT AVE'] # one complaint

23391    7500 S WOLCOTT AVE
Name: STREET ADDRESS, dtype: object

In [28]:
df4['Street Address'][df4['Street Address'] == '7500 S WOLCOTT AVE'] # one complaint

4296    7500 S WOLCOTT AVE
Name: Street Address, dtype: object

Based on the above analysis conducted on data only from the past year,
it shows that only one complaint regarding Pot Holes and Sanitation came from the address "7500 S WOLCOTT AVE".
Thus the probability that a complaint from this address will come regarding Pot Holes and Sanitation is 0.5 each.
The probability that a complaint from this address will come regarding Vacant Houses and Graffiti is zero (0).

Question 3(b)

In [79]:
lawndale_count = df2[df2['Street Address'].str.contains("LAWNDALE")]
lawndale_count['Street Address'].count()

617

In [97]:
#From Wikipedia: Uptown tows include: Foster Avenue, Lake Michigan, Montrose, Irving Park, Clark, Ravenswood
uptown_count = df2[df2['Street Address'].str.contains("FOSTER AVE|MONTROSE|IRVING PARK|CLARK|RAVENSWOOD")==True] # source: stackoverflow
uptown_count['Street Address'].count()

4395

If a given complaint regarding Graffiti is from Lawndale, its probability is 617/(617+4395)=0.123.
If a given complaint regarding Graffiti is from Uptown, its probability is 4395/(617+4395)=0.877.

Question 3(c)

Total Graffiti calls = 100 (Englewood) + 160 (Uptown) = 260.
Probability that the call came from Englewood = 100/260 = 0.385
Probability that the call came from Uptown = 160/260 = 0.615

It is about twice as likely that the call came from Uptown (0.615) than from Englewood (0.385).